var/spack/repos/builtin/packages/nekrs/package.py

```diff
@@ -0,0 +1,189 @@
+        # second time from OCCA - however, only the second one will be seen in
+        # the verbose output, so we keep both.
+        cxxflags = spec.compiler_flags['cxxflags']
+        if cxxflags:
+            env.set('CXXFLAGS', ' '.join(cxxflags))
+
+        # For the cuda, openmp, and opencl variants, set the environment
+        # variable OCCA_{CUDA,OPENMP,OPENCL}_ENABLED only if the variant is
+        # disabled. Otherwise, let OCCA autodetect what is available.
+
+        if '+cuda' in spec:
+            cuda_dir = spec['cuda'].prefix
+            cuda_libs_list = ['libcuda', 'libcudart', 'libOpenCL']
+            cuda_libs = find_libraries(cuda_libs_list,
+                                       cuda_dir,
+                                       shared=True,
+                                       recursive=True)
+            env.set('OCCA_INCLUDE_PATH', cuda_dir.include)
+            env.set('OCCA_LIBRARY_PATH', ':'.join(cuda_libs.directories))
+            env.set('OCCA_CUDA_ENABLED', '1')
+        else:
+            env.set('OCCA_CUDA_ENABLED', '0')
+
+        # Disable hip autodetection for now since it fails on some machines.
+        if '+hip' in spec:
+            env.set('OCCA_HIP_ENABLED', '1')
+        else:
+            env.set('OCCA_HIP_ENABLED', '0')
+
+        if '+opencl' in spec:
+            env.set('OCCA_OPENCL_ENABLED', '1')
+        else:
+            env.set('OCCA_OPENCL_ENABLED', '0')
+
+        # Setup run-time environment for testing.
+        env.set('OCCA_VERBOSE', '1')
+        self._setup_runtime_flags(env)
+
+    def setup_run_environment(self, env):
+        # The 'env' is included in the Spack generated module files.
+        self._setup_runtime_flags(env)
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        # Export OCCA_* variables for everyone using this package from within
+        # Spack.
+        self._setup_runtime_flags(env)
+
+    def install(self, spec, prefix):
+        script_dir = 'scripts'
+
+        cc  = spec['mpi'].mpicc
+        fc  = spec['mpi'].mpifc
+        cxx = spec['mpi'].mpicxx
+
+        fflags = spec.compiler_flags['fflags']
+        cflags = spec.compiler_flags['cflags']
+        cxxflags = spec.compiler_flags['cxxflags']
+
+        if self.compiler.name in ['xl', 'xl_r']:
+            # Use '-qextname' to add underscores.
+            # Use '-WF,-qnotrigraph' to fix an error about a string: '... ??'
+            fflags += ['-qextname', '-WF,-qnotrigraph']
+
+        error = Executable(fc)('empty.f', output=str, error=str,
+                               fail_on_error=False)
+
+        if 'gfortran' in error or 'GNU' in error or 'gfortran' in fc:
+            # Use '-std=legacy' to suppress an error that used to be a
+            # warning in previous versions of gfortran.
+            fflags += ['-std=legacy']
+
+        fflags = ' '.join(fflags)
+        cflags = ' '.join(cflags)
+        cxxflags = ' '.join(cxxflags)
```

Comment:
Your flags don't seem to be used as part of the actual build? I think 154 through 173 could justs be removed.

Reply:
I removed them from the package.